Source/Plugins/Core/com.equella.serverbase/src/com/tle/core/replicatedcache/dao/ReplicatedCacheDao.java

```diff
@@ -26,6 +26,8 @@
 public interface ReplicatedCacheDao extends GenericDao<CachedValue, Long> {
   CachedValue get(String cacheId, String key);
 
+  CachedValue getByValue(String cacheId, String value);
```

Comment:
This should have symmetry with `put()` and so should be:

```suggestion
  CachedValue getByValue(String cacheId, byte[] value);
```

That also then means you can keep the knowledge around Base64 encoding encapsulated.